src/borg/testsuite/archiver.py

```diff
@@ -988,6 +988,7 @@ def test_readonly_repository(self):
             # Restore permissions so shutil.rmtree is able to delete it
             os.system('chmod -R u+w ' + self.repository_path)
 
+    @pytest.mark.skipif('BORG_TESTS_UNRELIABLE_MODES' in os.environ, reason='modes unreliable')
```

Comment:
also, at least a short comment in the source about why we have this would be useful for the future reader.